modules/network/src/main/java/org/apache/ignite/internal/network/serialization/Primitives.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.network.serialization;
+
+/**
+ * Utils to work with primitives.
+ */
+public class Primitives {
+    /**
+     * Returns number of bytes a value of the given primtive type takes (1 for byte, 8 for long and so on).
+     *
+     * @param clazz primitive type
+     * @return number of bytes
+     * @throws IllegalArgumentException if the passed type is not primitive
+     */
+    public static int widthInBytes(Class<?> clazz) {
+        if (clazz == byte.class) {
+            return Byte.BYTES;
+        } else if (clazz == short.class) {
+            return Short.BYTES;
+        } else if (clazz == int.class) {
+            return Integer.BYTES;
```

Comment:
I'm just curious if this won't work for varints

Reply:
What do you mean? Elaborate please. Here, we just cover all the primitive types defined in the JLS.

Reply:
Ok, so these values are not used for serialization/de-serialization?

Reply:
They are just used to compute correct offsets into the 'primitive data' byte array, so they must represent the real 'width' of types in accordance with the JLS